lib/plugins/ray_trace.js

```diff
@@ -23,4 +24,41 @@ module.exports = (bot) => {
 
     return bot.world.raycast(eyePosition, viewDirection, maxDistance, matcher)
   }
+
+  bot.entityAtCursor = (maxDistance) => {
+    maxDistance = maxDistance ?? 3.5
+    const block = bot.blockAtCursor(maxDistance)
+    maxDistance = block?.intersect.distanceTo(bot.entity.position) ?? maxDistance
+
+    const entities = Object.values(bot.entities)
+      .filter(entity => entity.type !== 'object' && entity.username !== bot.username && entity.position.distanceSquared(bot.entity.position) <= maxDistance ** 2)
+
+    const dir = new Vec3(-Math.sin(bot.entity.yaw) * Math.cos(bot.entity.pitch), Math.sin(bot.entity.pitch), -Math.cos(bot.entity.yaw) * Math.cos(bot.entity.pitch))
+    const iterator = new RaycastIterator(bot.entity.position.offset(0, bot.entity.height, 0), dir.normalize(), maxDistance)
+
+    const intersectArr = []
+
+    let pos = iterator.next()
+    while (pos) {
+      for (let i = 0; i < entities.length; i++) {
```

Comment:
you are still testing all entities for each block, this loop needs to be outside the while